setup_unit_tests.py

```diff
@@ -0,0 +1,488 @@
+            "To run the unit tests, see [insert other script here]"
+        )
+    )
+    parser.add_argument(
+        "--version",
+        "-v",
+        default="1.1",
+        help="Version of the spec to grab unit tests out of. Only WDL 1.1 and above has support for built in unit tests.",
+        choices=["1.1", "1.2"]
+    )
+    parser.add_argument(
+        "--output-type",
+        default="yaml",
+        choices=["yaml", "json"],
+        help="Specify conformance file output type."
+    )
+    parser.add_argument(
+        "--force-pull",
+        default=False,
+        action="store_true",
+        help="Default behavior does not pull the spec repo if it was already pulled. Specify this argument to force a refresh."
+    )
+    parser.add_argument(
+        "--extra-metadata", "-e",
+        default="unit_tests_metadata.yaml",
+        help="Include extra metadata when extracting the unit tests into the config. "
+             "Since we have our own method of testing file outputs, this is mostly used for file hashes/regexes."
+    )
+    parser.add_argument(
+        "--script",
+        # default="unit_tests_script.sh",
+        default=None,
+        help="Bash script to run alongside. This is to set up the environment for the test suite, ex mount points and files tests may depend on."
+             "(Probably need to run with root)."
+    )
+    parser.add_argument(
+        "--repo",
+        # while the WDL spec has its bugs, use a fixed version
+        # see openwdl issues #653, #654, #661, #662, #663, #664, #665, #666
+        default="https://github.com/stxue1/wdl.git",
+        help="Repository to pull from."
+    )
+    argcomplete.autocomplete(parser)
+    args = parser.parse_args(argv)
+
+    spec_dir = f"wdl-{args.version}-spec"
+    if not os.path.exists(spec_dir) or args.force_pull is True:
+        print(f"Pulling SPEC from repo {args.repo}...")
+        cmd = f"git clone {args.repo}"
+        subprocess.run(cmd.split(), stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+
+    os.chdir(spec_dir)
+
+    # may be fragile if WDL changes their branch naming scheme
+    # todo: remove -fixes suffix after wdl fixes their spec, see comment above
+    cmd = f"git checkout wdl-{args.version}-fixes"
+    subprocess.run(cmd.split(), stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+
+    os.chdir("..")
+
+    # temp
+    cmd = f"rm -rf unit_tests"
+    subprocess.run(cmd.split(), stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+
+    print("Extracting tests...")
+    extract_tests(Path(spec_dir) / Path("SPEC.md"), Path(spec_dir) / Path("tests/data"), Path("unit_tests"), args.version, args.output_type, args.extra_metadata)
+
+    if args.script is not None:
+        # assume it needs to run as root
+        if os.geteuid() == 0:
+            subprocess.run(["bash", "{args.script}"], shell=True)
+        else:
+            subprocess.run(["sudo", "bash", "{args.script}"], shell=True)
```

Comment:
Don't we anticipate extracting the tests on one machine or on one day, and then later coming back and running the extracted tests? Doesn't this script need to run before the tests are *run*, and not after they are extracted?

Reply:
I changed it to run before running the unit tests. I still left it empty by default as this script is only to make one of the unit tests work.